### PR TITLE
Quick search

### PR DESCRIPTION
**Pull request by arunaousula9421**
@seanculleton Please Review

### CODE REVIEW

File: property/src/pages/home.page.ts [hunk 1]
```diff
@@ -16,4 +16,73 @@ export class HomePage extends Page {
   public navBarComponet(): NavBar {
     return new NavBar();
   }
+
+  public get quickSearchBlock() {
+    return browser.element('.search-block');
+  }
+
+  public get moreOptions() {
+    return browser.element('.//*[@ng-show="headingBox"]/li/a[text()="More"]');
```
Comment (seanculleton) [inline property/src/pages/home.page.ts]: Should this locator here start with .//? I think it should just be //

File: property/src/pages/home.page.ts [hunk 2]
```diff
@@ -16,4 +16,73 @@ export class HomePage extends Page {
...
+  }
+
+  public get cityCountyOptions() {
+    return browser.element('.//*[@class="cell"][1]');
```
Comment (seanculleton) [inline property/src/pages/home.page.ts]: same here

File: property/src/pages/home.page.ts [hunk 3]
```diff
@@ -16,4 +16,73 @@ export class HomePage extends Page {
...
+  }
+
+  public get areaOptions() {
+    return browser.element('.//*[@class="cell"][2]');
```
Comment (seanculleton) [inline property/src/pages/home.page.ts]: And here

File: property/src/pages/home.page.ts [hunk 4]
```diff
@@ -16,4 +16,73 @@ export class HomePage extends Page {
...
+  }
+
+  public get dublinCityCentre() {
+    return browser.element(
+      './/*[@class="ng-binding ng-scope"][text()="Dublin City Centre"]'
+    );
+  }
+
+  public get searchBtn() {
+    return browser.element('.//*[@class="btn btn-primary btn-search"]');
```
Comment (seanculleton) [inline property/src/pages/home.page.ts]: and here

File: property/src/pages/home.page.ts [hunk 5]
```diff
@@ -16,4 +16,73 @@ export class HomePage extends Page {
...
+  }
+
+  public verifySearchBlockIsVisible() {
+    return this.quickSearchBlock.isVisible();
+  }
+
+  public selectCategory(Category: String) {
+    var visibleCategoryTypes = browser.element(
+      './/*[@ng-show="headingBox"]/li/a[text()="' + Category + '"]'
```
Comment (seanculleton) [inline property/src/pages/home.page.ts]: and here

File: property/src/pages/home.page.ts (hunk 1 above)
Comment (arunaousula9421) [reply property/src/pages/home.page.ts]: Yes we can do that .. It was just copy and paste from firefox ..

File: property/src/pages/home.page.ts [hunk 6]
```diff
@@ -16,4 +16,73 @@ export class HomePage extends Page {
...
+    );
+    var hiddenCategoryTypes = browser.element(
+      './/*[@ng-hide="headingBox"]/li/a[text()="' + Category + '"]'
```
Comment (seanculleton) [inline property/src/pages/home.page.ts]: and here

File: property/src/pages/home.page.ts [hunk 7]
```diff
@@ -16,4 +16,73 @@ export class HomePage extends Page {
...
+    return browser.element('.//*[@class="cell"][2]');
+  }
+
+  public get dublinCity() {
+    return browser.element('//li[*]/span[text()="Dublin City"]');
+  }
+
+  public get dublinCityCentre() {
+    return browser.element(
+      './/*[@class="ng-binding ng-scope"][text()="Dublin City Centre"]'
```
Comment (seanculleton) [inline property/src/pages/home.page.ts]: and here

**Pull request merged**